.github/workflows/build.yml

```diff
@@ -89,9 +89,18 @@ jobs:
           GITHUB_ACTOR: ${{ github.actor }}
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+      # Run Maven Deploy (if either running on main or a version tag on a fork)
```

Comment:
As forks are potentially ephemeral upstream repository won't care about forks. You shall not implement fork specific steps into your workflows and push to the upstream repository, even if it's a catenax-ng fork. If you want to run workflows in your fork, then you have to mange it otherwize.